File: sockets/src/serialization/SerializationSimpleTCPClient.java
package serialization;

import java.io.*;
import java.net.*;

class SerializationSimpleTCPClient {
	public static void main(String argv[]) {

		Pessoa p1 = new Pessoa("Joao da Silva", "Quixada", 999881122, 2012);
		
		System.out.println("serialization.Pessoa p1: " + p1.toString());
		Socket clientSocket;
		try {
			clientSocket = new Socket("localhost", 6789);
			ObjectOutputStream outToServer = new ObjectOutputStream(clientSocket.getOutputStream());
			ObjectInputStream inFromServer = new ObjectInputStream(clientSocket.getInputStream());
			outToServer.writeObject(p1);
			Pessoa p2 = (Pessoa) inFromServer.readObject();
			System.out.println("serialization.Pessoa p2: " + p2.toString());
			clientSocket.close();
		} catch (IOException e) {
			e.printStackTrace();
		} catch (ClassNotFoundException e) {
			e.printStackTrace();
		}
	}
}

File: sockets/python/serialization/serializationsimpletcpclient.py
import socket, pickle

from pessoa import Pessoa
HOST = 'localhost'
PORT = 6789
s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
s.connect((HOST, PORT))
pessoa = Pessoa("Joao da Silva", 'Quixada', '999990000', '2014')
print(pessoa.toString())
data_string = pickle.dumps(pessoa)
s.sendall(data_string)
data = s.recv(1024)
pessoa = pickle.loads(data)
print(pessoa.toString())
s.close()